src/pipe.cpp

```diff
@@ -615,3 +615,19 @@ void zmq::pipe_t::set_disconnect_msg (
       _disconnect_msg.init_buffer (&disconnect_[0], disconnect_.size ());
     errno_assert (rc == 0);
 }
+
+void zmq::pipe_t::send_hiccup_msg (const std::vector<unsigned char> &hiccup_)
+{   
+    if (!hiccup_.empty () && _out_pipe) {
+        msg_t msg;
+        const int rc = msg.init_buffer (&hiccup_[0], hiccup_.size ());
+        errno_assert (rc == 0);
+ 
+        // Rollback any incomplete message in the pipe, and push the disconnect message.
```

Comment:
Should this behaviour be documented in the manpage? It results in lost (half-received) messages, right?

Reply:
no, because those messages were not flushed yet. Actually, it is not needed, as `clean_pipes` already did that.